modules/pytket-cirq/pytket/extensions/cirq/cirq_convert.py

```diff
@@ -182,13 +182,17 @@ def cirq_to_tk(circuit: cirq.circuits.Circuit) -> Circuit:
     return tkcirc
 
 
-def tk_to_cirq(tkcirc: Circuit) -> cirq.circuits.Circuit:
+def tk_to_cirq(tkcirc: Circuit, copy_all_qubits: bool = False) -> cirq.circuits.Circuit:
     """Converts a tket :py:class:`Circuit` object to a Cirq :py:class:`Circuit`.
 
     :param tkcirc: The input tket :py:class:`Circuit`
 
     :return: The Cirq :py:class:`Circuit` corresponding to the input circuit
     """
+    if copy_all_qubits:
+        for q in tkcirc.qubits:
+            tkcirc.add_gate(OpType.noop, [q])
```

Comment:
This will modify the circuit passed in, which seems undesirable, so I think in this case we should take a copy. I think it should suffice to put `tkcirc = tkcirc.copy()` before the line `for q in tkcirc.qubits:` (but a test to verify that this method doesn't modify the original circuit would be good).

Reply:
I've added a copy as suggested.
I've extended the test_conversions test to make sure that this copy behaviour is there (confirming that the converted circuit doesn't match the original circuit, i.e. the original is unchanged).